### PR TITLE
feat: Start updating site for Docker

### PR DESCRIPTION
**Pull request by darcywong00**
This starts to address keymanapp/keyman.com#337 for s.keyman.com

Based on the Docker work used in help.keyman-staging.com
* Commented out composer-related steps
* Migrated mime types from web.config

Currently have not:
* migrated the Kudu webjob .bat file 
* handle the K8s files

### CODE REVIEW

File: .htaccess [hunk 1]
```diff
@@ -1,6 +1,18 @@
...
+    AddType font/otf .otf
+    AddType application/octet-stream .eot
+    AddType application/x-apple-aspen-config .mobileconfig
+    AddType image/svg+xml .svg
+    AddType application/javascript .js 
+
+    AddCharset UTF-8 .js
+</IfModule>
 <IfModule mod_rewrite.c>
     RewriteEngine on
```
Comment (darcywong00) [inline .htaccess:18]: RewriteRule on l.19
Should we have `build.sh` and `.htaccess` give a 404?

**Comment by darcywong00** [comment]
> Currently have not:
>
>migrated the Kudu webjob .bat file
>handle the K8s files

@tim-eves / @mcdurdin 
Do you know if the [kudu job](https://github.com/keymanapp/s.keyman.com/tree/master/App_Data/jobs/continuous/deploy) will be replaced by K8s?

**Comment by mcdurdin** [comment]
> Do you know if the [kudu job](https://github.com/keymanapp/s.keyman.com/tree/master/App_Data/jobs/continuous/deploy) will be replaced by K8s?

Yes, we need an alternate deployment method -- this one is not suitable for Docker

File: .htaccess (hunk 1 above)
Comment (mcdurdin) [reply .htaccess:18]: yes good idea

File: resources/keyman-site.conf [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<FilesMatch .md$>
+	SetHandler text/html
+</FilesMatch>
+
+DirectoryIndex index.md index.php index.html
```
Comment (mcdurdin) [inline resources/keyman-site.conf:5]: These shouldn't be necessary for s.keyman.com?

File: resources/keyman-site.conf [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+<Directory /var/www/html>                
+	Options +Includes +FollowSymLinks -MultiViews
+	AllowOverride All
+</Directory>
+
+php_value include_path "/var/www/html/_includes:."
```
Comment (mcdurdin) [inline resources/keyman-site.conf:12]: This shouldn't be necessary as s.keyman.com is very lightweight and none of the PHP files should have includes.

We should also be removing the api/deploy.php, api/webhooks.php and App_Data/jobs/continuous/deploy/* files as they won't be needed in the docker setup. Presumably the .github/workflows will need redesign as well.

And then the only .php file of relevance is api/kmwversion.php.

**Review by mcdurdin — CHANGES_REQUESTED** [review]

File: resources/keyman-site.conf (hunk 3 above)
Comment (darcywong00) [reply resources/keyman-site.conf]: Still TODO on the .github/workflow

**Comment by darcywong00** [comment]
Updated to change the base branch to `staging`

File: .htaccess [hunk 4]
```diff
@@ -1,8 +1,20 @@
 <IfModule mod_headers.c>
     Header set Access-Control-Allow-Origin "*"
 </IfModule>
+<IfModule mod_mime.c>
+    AddType font/ttf .ttf
+    AddType font/woff .woff
+    AddType font/woff2 .woff2
+    AddType font/otf .otf
+    AddType application/octet-stream .eot
```
Comment (mcdurdin) [inline .htaccess:9]: Could do:
```suggestion
    AddType application/vnd.ms-fontobject .eot
```

Per https://www.iana.org/assignments/media-types/application/vnd.ms-fontobject

File: .htaccess [hunk 5]
```diff
@@ -1,8 +1,20 @@
...
+    AddType image/svg+xml .svg
+    AddType application/javascript .js 
+
+    AddCharset UTF-8 .js
+</IfModule>
 <IfModule mod_rewrite.c>
     RewriteEngine on
 
-    RewriteRule ^(cdnrefresh.php|Dockerfile|web.config|.git|App_Data)$ - [R=404,L]
+    RewriteRule ^(build.sh|cdnrefresh.php|Dockerfile|web.config|.git|.htaccess|App_Data)$ - [R=404,L]
```
Comment (mcdurdin) [inline .htaccess:19]:
```suggestion
    RewriteRule ^(build.sh|Dockerfile|.git|resources/)$ - [R=404,L]
```

* I believe .htaccess will always be [blocked](https://stackoverflow.com/a/36482767/1836776) by Apache default setup.
* cdnrefresh.php is gone
* We should also delete web.config from this branch.
* We should also delete .github from this branch.
* App_Data will also go in its entirety
* resources/ folder should not be accessible

File: resources/keyman-site.conf
Comment (mcdurdin) [inline resources/keyman-site.conf:1]: I'm not sure we need this file at all? Which would mean removing it from the Dockerfile as well.

**Review by mcdurdin — CHANGES_REQUESTED** [review]

**Review by mcdurdin — APPROVED** [review]

**Pull request merged**